### DISCUSSION TITLE
question about backing up gh repos to vlab

### DISCUSSION THREAD

**Discussion by k-doering-NOAA** [nmfs-stock-synthesis/stock-synthesis#423] [Team Posts]
Hi all, I'm trying out "teams discussions" for our "developers" team. This might be a good way to discuss ss related things that users don't need to get noifications for (although I think everyone on the developers team will see it).

Anyway, I'm curious about how we should be doing our backup to vlab? Right now, we are maintaining mirrored repositories for the stock synthesis source code, documentation, and gui, but a method that Corinne does is just zip the files from the repo weekly and send the zipped repositories to vlab.

Which should we be doing for our repositories in this org? Do we want the back up to be fully usable (i.e., we could start pushing to vlab instead of github if github goes down or browse files  at any time), or do we just want a zipped back up of the repository, which would mostly be useful for meeting the back up requirement and in case we did need a backup because our repo ceased to exist on github?

**Comment by Rick-Methot-NOAA** [comment]
What does the NWFSC guidance recommend?  We definitely need to do something.

**Comment by k-doering-NOAA** [comment]
I think either approach would satisfy the requirements. Another option is potentially having NWFSC IT do it for us, backing up to a NWFSC resource. I'm not sure who to ask about this, but could look into this more.

**Comment by chantelwetzel-noaa** [comment]
If there is no significant overhead or effort required in our current
approach, mirror between github and vlab, that would be my preference.
That way if there ever was an issue with github we would not lose any work.

Chantel Wetzel, PhD

*Pronouns: she/her (what does that mean? <https://www.mypronouns.org/>)*

*Fishery Resource Analysis and Monitoring DivisionNorthwest Fisheries
Science Center*
*National Marine Fisheries Service*
*National Oceanic and Atmospheric Administration*
*Office: 206.302.1753 *

**Comment by iantaylor-NOAA** [comment]
I thought that the NWFSC system was to list your repo on this spreadsheet
https://docs.google.com/spreadsheets/d/1bSDo6qG7XEPZnRVPpxhtg2bozY_3Lqg8ewydsQONfFk/edit#gid=704587532
and it would automatically get backed up.

But I see on this site
https://archive.fisheries.noaa.gov/nwfsc/research/datatech/data/source_code.html
that the nwfsc-assess repositories are backed up even though they never got added to the spreadsheet.
That public backup site says it was last updated in June, but the github policy indicates that there's a private backup on the NWFSC internal gitlab server which presumably gets updated more often.

**Comment by k-doering-NOAA** [comment]
@chantelwetzel-noaa , yeah, I think the current mirroring takes about as much work (maybe less commands, actually) than zipping the repos to back them up. The only disadvantage I see is someone could accidentally push to vlab repos - but this is easily sorted.

@iantaylor-NOAA , thanks, that is more details than I knew!

Maybe the right way forward is to continue mirroring the repositories. We also have created some new ones (ss-flowcharts, user-examples) that I can start mirroring). Does that sound ok?
